### ISSUE TITLE
AWS ELB support for both v2 elbs (network & application elb)

### ISSUE THREAD

**Issue by stefanhorning** [ansible/ansible#38094]
##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME
elb_application_lb

##### SUMMARY
AWS has two generations for ELB (ElasticLoadbalancing) products. V1, also referred to as ELB classic and v2.

###### AWS ELB situation

In **v1** there is one type of ELB that can be created, I would call classic ELB. Ansible already has a module named like that called `elb_classic_lb`.

In **v2** however there is **two** types of ELBs that can be created. One is called `application` and the other `network`. But Ansible has a module called `elb_application_lb` only. By looking at Boto3 one can see that `Type` is just a parameter for the v2 API. See https://boto3.readthedocs.io/en/latest/reference/services/elbv2.html#ElasticLoadBalancingv2.Client.create_load_balancer

###### Ansible approach

So accordingly the module should be called `elb_v2_lb` with a parameter to switch between `network` and `application`. Or there should be two modules (probably just wrapers arround the same class) called `elb_application_lb` and `elb_network_lb`. Not sure what the best approach is here, but things need changing either way. 

I would be willing to support work towards supporting both types. But before starting any PR we should solve the question first on which way to go here.

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/cloud/amazon/elb_application_lb.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/elb_application_lb.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @willthames @wimnat
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by wimnat** [comment]
Hi @stefanhorning. Some work has already gone in to a network load balancer module. PR can be found here https://github.com/ansible/ansible/pull/33808. 

There is another PR pending for changes to the application lb module to help support this. 

I'm hopeful that both will be merged soon.

**Comment by stefanhorning** [comment]
Perfect, thanks. I was looking for s.th. like this but nothing showed up in my searches. It looks like you already implemented exactly what I layed out here in this issue. So I will close this issue.
